src/main/java/org/wso2/carbon/connector/operations/BatchExecute.java

```diff
@@ -0,0 +1,153 @@
+
+    @Override
+    public void connect(MessageContext messageContext) {
+        ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
+        String connectionName = null;
+        SnowflakeConnection snowflakeConnection = null;
+
+        try {
+            connectionName = SnowflakeUtils.getConnectionName(messageContext);
+            snowflakeConnection = (SnowflakeConnection) handler.getConnection(Constants.CONNECTOR_NAME, connectionName);
+            SnowflakesOperationResult result = batchExecuteQuery(messageContext, snowflakeConnection);
+            SnowflakeUtils.setResultAsPayload(messageContext, result);
+        } catch (InvalidConfigurationException e) {
+            handleError(messageContext, e, Error.INVALID_CONFIGURATION, ERROR_MESSAGE);
+        } catch (SnowflakeOperationException e) {
+            handleError(messageContext, e, Error.OPERATION_ERROR, ERROR_MESSAGE);
+        } catch (ConnectException e) {
+            handleError(messageContext, e, Error.CONNECTION_ERROR, ERROR_MESSAGE);
+        } finally {
+            if (snowflakeConnection != null) {
+                handler.returnConnection(Constants.CONNECTOR_NAME, connectionName, snowflakeConnection);
+            }
+        }
+    }
+
+    /**
+     * Executes the query.
+     *
+     * @param messageContext      Message Context.
+     * @param snowflakeConnection Snowflake Connection.
+     * @return SnowflakesOperationResult.
+     * @throws InvalidConfigurationException Invalid Configuration Exception.
+     * @throws SnowflakeOperationException  Snowflake Operation Exception.
+     */
+    private SnowflakesOperationResult batchExecuteQuery(MessageContext messageContext, SnowflakeConnection snowflakeConnection)
+            throws InvalidConfigurationException, SnowflakeOperationException {
+        SnowflakesOperationResult snowflakesOperationResult =
+                new SnowflakesOperationResult(OPERATION_NAME, false);
+        String query = (String) getParameter(messageContext, Constants.EXECUTE_QUERY);
+        String payload = (String) getParameter(messageContext, Constants.PAYLOAD);
+
+        if (StringUtils.isEmpty(query)) {
+            throw new InvalidConfigurationException("Execute Query is not provided.");
+        }
+
+        if (StringUtils.isEmpty(payload)) {
+            throw new InvalidConfigurationException("Empty Payload is provided.");
+        }
+
+        JsonArray executeArray = JsonParser.parseString(payload).getAsJsonArray();
+        String[] columns = SnowflakeUtils.getColumnNames(query);
+        PreparedStatement preparedStatement = null;
+
+        try {
+            preparedStatement = snowflakeConnection.getConnection().prepareStatement(query);
+            if (executeArray.size()>0 && columns.length>0) {
+                for (int i = 0; i<executeArray.size(); i++) {
+                    JsonObject executeObject = executeArray.get(i).getAsJsonObject();
```

Comment:
what if we only need to add single column. in that case input is a string array, not object array